### PR TITLE
Added logic to filter noise and consider only majority light change

### PR DESCRIPTION
**Pull request by arunsuresh16**

### CODE REVIEW

File: Self_Driving_Car_Engineer/Capstone_Project/CarND-Capstone/ros/src/tl_detector/light_classification/tl_classifier.py [hunk 1]
```diff
@@ -110,49 +136,28 @@ def get_classification(self, image):
...
-            elif final_classes[0] == 3:
-                return TrafficLight.RED
-            
-        
-        return TrafficLight.UNKNOWN
+            # Filter boxes with a confidence score less than CONFIDENCE_CUTOFF
+            final_boxes, final_scores, final_classes = self.filter_boxes(CONFIDENCE_CUTOFF, boxes, scores, classes)
+
+            if len(final_boxes) > 0:
+                state, final_boxes, final_classes = self.find_state_change_and_noise_filter(final_boxes, final_classes)
```
Comment (zackZhu1) [inline Self_Driving_Car_Engineer/Capstone_Project/CarND-Capstone/ros/src/tl_detector/light_classification/tl_classifier.py]: where do we use this state?

Comment (arunsuresh16) [reply Self_Driving_Car_Engineer/Capstone_Project/CarND-Capstone/ros/src/tl_detector/light_classification/tl_classifier.py]: good point. I was using that in the notebook. Now its not required

Comment (zackZhu1) [reply Self_Driving_Car_Engineer/Capstone_Project/CarND-Capstone/ros/src/tl_detector/light_classification/tl_classifier.py]: @arunsuresh16

**Review by vzebilis — APPROVED** [review]
Thank you for the changes Arun!
They seem just fine. As a minor enhancement we could add a a confidence check as a tiebreaker when we detect an equal number of boxes for different classes in an image.
This can be added later if needed.

**Review by zackZhu1 — APPROVED** [review]

**Pull request merged**